### PR TITLE
Fix error in 'ensure_consistent_bilou_tagging'

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Fix error in `ensure_consistent_bilou_tagging`.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/utils/bilou_utils.py [hunk 1]
```diff
@@ -295,6 +295,15 @@ def _find_bilou_end(start_idx: int, predicted_tags: List[Text]) -> int:
     start_tag = tag_without_prefix(predicted_tags[start_idx])
 
     while not finished:
+        if current_idx >= len(predicted_tags):
+            logger.debug(
+                "Inconsistent BILOU tagging found, B- tag not closed by L- tag, "
+                "i.e [B-a, I-a, O] instead of [B-a, L-a, O].\n"
+                "Assuming last tag is L- instead of I-."
```
Comment (akelad) [inline rasa/nlu/utils/bilou_utils.py:302]: @tabergma in what type of situations does this actually happen?

Comment (tabergma) [reply rasa/nlu/utils/bilou_utils.py:302]: `CRFEntityExtractor` or `DIETClassifier` may predict inconsistent BILOU tags.

**Comment by akelad** [comment]
thanks for the fix :D should we release a patch directly?

**Comment by tabergma** [comment]
yes, that was my plan, will do a patch release now

**Pull request merged**